### PR TITLE
Fix slice expression type

### LINKED ISSUE #1886: The optional endIndex of the slice expression is missing in types.
**maplibre-gl-js version**: 2.4.0

### Steps to Trigger Behavior

 1. Use the [`slice`](https://maplibre.org/maplibre-gl-js-docs/style-spec/expressions/#slice) expression anywhere.
 2. Add an end index somewhere.
 3. TypeScript does not accept it.

### Expected Behavior

The optional second index of the [`slice`](https://maplibre.org/maplibre-gl-js-docs/style-spec/expressions/#slice) expression should be accepted. It actually works.

### Actual Behavior

While it does work, TypeScript rejects it as the second optional `index` is not included in the types. A workaround is to use `@ts-ignore`

### PR DESCRIPTION
**Pull request by kevinschaul**
Builds on PR #1889 by adding a few tests.

Fixes #1886

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Write tests for all new functionality.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 - Make request for ImageSource cancelable ([#1802](https://github.com/maplibre/maplibre-gl-js/pull/1802))
 
 ### 🐞 Bug fixes
+- Fix the `slice` expression type ([#1886](https://github.com/maplibre/maplibre-gl-js/issues/1886))
```
Comment (HarelM) [inline CHANGELOG.md:21]: This should be in the top most bug section.

**Comment by HarelM** [comment]
This looks great, thanks!!
Just move the change log entry up and I'll merge this.

File: CHANGELOG.md (hunk 1 above)
Comment (kevinschaul) [reply CHANGELOG.md]: Fixed!

**Review by HarelM — APPROVED** [review]

**Comment by HarelM** [comment]
THANKS!!

**Pull request merged**